src/pages/admin/index.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import Router from 'next/router';
+import { useSession } from 'next-auth/react';
+import { QRGenerator } from '../../components/App';
+
+const Admin = () => {
+  const { data: session, status } = useSession();
+
+  React.useEffect(() => {
+    if (status === 'unauthenticated') {
+      Router.push('/auth/login');
+    }
+  }, [status]);
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  const email = session?.user?.email;
+  if (status === 'authenticated' && email.split('@')[1] === 'ubc.ca') {
```

Comment:
Why does the email have to end with "ubc.ca"? And if it does, it should be a constant defined globally elsewhere.
Like we should have permission functions like `isAdminUser()` that return booleans and the logic of defining an admin user should be there